docs/modules/data-structures/pages/working-with-external-data.adoc

```diff
@@ -250,6 +250,106 @@ include::ROOT:example$/dds/map/PersonEntryStore.java[tag=personms]
 
 TIP: link:https://github.com/hazelcast/hazelcast-code-samples/tree/master/distributed-map/mapstore/src/main/java[See more MapStore/MapLoader code samples].
 
+== Offloading MapStore and MapLoader Operations
+
+To improve cluster throughput, all MapStore operations do not block partition threads.
+Each partition in a member is managed by a single thread, ensuring that the write operations to a given partition are handled
+one at a time in first-in-first-out order. However, MapStore operations may take a long time to connect to the external data source and complete, blocking the affected partitions. 
+To avoid blocking partition threads, MapStore operations that communicate with the external data source are handled asynchronously, using an <<configuring-the-offload-executor, executor>>.
+
+=== Limitations
+
+* The `init()` and `destroy()` methods of the `MapLoaderLifecycleSupport` interface cannot be offloaded. These methods are triggered only once in the whole lifecycle of a map.
+* MapStore methods `loadAll()` and `loadAllKeys()` cannot be offloaded.
```

Comment:
>MapStore methods `loadAll()` and `loadAllKeys()` cannot be offloaded.

These methods are also offloaded independent of the work done in 5.2. In 5.1, they are also working on offloaded manner.